**Import required libraries**

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

**Read the dataset**

In [2]:
df=pd.read_csv('/content/StudentsPerformance.csv')

**Find out how many males and females participated in the test**

In [3]:
df[df.gender=='female'].gender.count()


518

518 females participated in the test

In [4]:
df[df.gender=='male'].gender.count()

482

482 males participated in the test

**Student's parental level of education**

In [5]:
df['parental level of education'].value_counts()

some college          226
associate's degree    222
high school           196
some high school      179
bachelor's degree     118
master's degree        59
Name: parental level of education, dtype: int64

Among parents of the students. Parents with college level of education is prominent

**Who scores the most on average for math,reading and writing based on gender**

In [6]:
df.groupby('gender')[['math score','reading score','writing score']].mean()

math score  reading score  writing score
gender                                          
female   63.376448      72.590734      72.467181
male     68.821577      65.545643      63.446058

male has more  math score than female. female has more reading and writing  score than male.

**Who scores the most on average for math, reading and writing based on test preparation course**

In [7]:
df.groupby('test preparation course')[['math score','reading score','writing score']].mean()

math score  reading score  writing score
test preparation course                                          
completed                  69.96648      74.175978      74.684358
none                       63.78972      66.417445      64.457944

math score,reading and writing score average is more to the students who completed test preparation course

**Scoring variation for math,reading and writing based on gender**

In [8]:
df.groupby('gender')[['math score','reading score','writing score']].var()

math score  reading score  writing score
gender                                          
female  256.958593     207.677438     220.369327
male    211.889097     200.211010     202.413924

Scoring variation of female in math score, reading and writing score is more than male

**Scoring variation for math,reading and writing based on test preparation course**

In [9]:
df.groupby('test preparation course')[['math score','reading score','writing score']].var()

math score  reading score  writing score
test preparation course                                          
completed                210.884027     183.265864     175.202612
none                     246.668662     213.419851     226.251739

math score,reading and writing score variation  is more to the students who does not completed test preparation course

**Find out top 25% of students based on their math  score**

In [10]:
np.quantile(df['math score'],0.75)

77.0

In [11]:
df[df['math score']>=77].sort_values(by='math score',ascending=False)

gender race/ethnicity parental level of education         lunch  \
999    male        group E           bachelor's degree      standard   
998    male        group E          associate's degree  free/reduced   
997    male        group D                some college      standard   
996    male        group A                some college      standard   
517  female        group E          associate's degree      standard   
..      ...            ...                         ...           ...   
854    male        group D            some high school      standard   
855    male        group E          associate's degree  free/reduced   
856    male        group E            some high school      standard   
415  female        group B             master's degree  free/reduced   
414  female        group A                 high school  free/reduced   

    test preparation course  math score  reading score  writing score  
999               completed         100            100            100  
998               completed         100            100             93  
997               completed         100             97             99  
996               completed         100             96             86  
517                    none         100            100            100  
..                      ...         ...            ...            ...  
854               completed          77             68             69  
855               completed          77             69             68  
856               completed          77             76             77  
415               completed          77             97             94  
414               completed          77             88             85  

[255 rows x 8 columns]

 First we are calculating the third quartile .Then  sort students with maths score greater than or equal to  77 in descending order.